doc/query.md

```diff
@@ -72,9 +105,47 @@ if (name == null) {
 }
 ```
 
-`ensureIndexed(List<Object> fieldNames, String indexName)` returns the name of the index if it is successful, otherwise it returns `null`.
+Since text search relies on SQLite FTS, we must ensure that SQLite was compiled with both the `ENABLE_FTS3` and `ENABLE_FTS3_PARENTHESIS` options.  To verify that text search is enabled and that a text index can be created use `isTextSearchEnabled()` before attempting to create a text index.  If text search is not enabled see [compiling and enabling SQLite FTS][enableFTS] and [SQLite Android Bindings][androidBind] for details. 
 
-If an index needs to be changed, first delete the existing index by calling `deleteIndexNamed(String indexName)` where the argument is the index name, then call  `ensureIndexed(List<Object> fieldNames, String indexName)` with the new definition.
+[enableFTS]: http://www.sqlite.org/fts3.html#section_2
+[androidBind]: https://www.sqlite.org/android/doc/trunk/www/index.wiki
+
+```java
+if (im.isTextSearchEnabled()) {
+    // Create a text index over the name and comment fields.
+    String name = im.ensureIndexed(Arrays.<Object>asList("name", "comment"), 
+                                   "basic_text_index", 
+                                   "text");
+    if (name == null) {
+        // there was an error creating the index
+    }
+}
+```
+
+Since text indexing relies on SQLite FTS functionality any custom tokenizers need to be managed through SQLite.  SQLite comes standard with the "simple" default tokenizer as well as a Porter stemming algorithm tokenizer ("porter").  Please refer to [SQLite FTS tokenizers][fts] for additional information on custom tokenizers.
+
+[fts]: http://www.sqlite.org/fts3.html#tokenizer  
+
+When creating a text index, overriding the default tokenizer setting is done by providing a `tokenize` parameter setting as part of the index settings.  In the example below we set the tokenizer to `porter`.
```

Comment:
Note that the value provided here is the name of the tokeniser given to SQLite when registering it.